velox/type/ShortDecimal.h

```diff
@@ -23,11 +23,14 @@
 
 namespace facebook::velox {
 
+struct LongDecimal;
+
 struct ShortDecimal {
  public:
   // Default required for creating vector with NULL values.
   ShortDecimal() = default;
-  constexpr explicit ShortDecimal(int64_t value) : unscaledValue_(value) {}
+  constexpr ShortDecimal(int64_t value) : unscaledValue_(value) {}
```

Comment:
I think you still need 'explicit'

Reply:
My idea was to allow all integer types to implicitly cast to ShortDecimal and LongDecimal.
I can add explicit constructors if that is a best practice.

Reply:
Why would we want to implicitly cast any integer to a decimal? XxxDecimal classes are a bit weird as they hold unscaled values which cannot be properly interpreted without extract information about scale. Implicitly casing an integer to XxxDecimal produces a value that is not equal to the original integer unless scale of 0 is assumed. I wonder if renaming these classes to UnscaledXxxDecimal would make the code a bit clearer.

Reply:
My interpretation is that in the input to XxxDecimal are always unscaled values as there is no precision, scale info like you said. But I see your view as well. I guess renaming to UnscaledXxxDecimal clarifies this and we can allow implicit integer cast to UnscaledXxxDecimal.
I will make this change in a separate PR.

Reply:
CC: @karteekmurthys